src/player_activity.cpp

```diff
@@ -242,11 +243,11 @@ void player_activity::do_turn( player *p )
             break;
         default:
             // Based on speed, not time
-            if( p->moves <= moves_left ) {
-                moves_left -= p->moves;
+            if( p->moves * season_mult <= moves_left ) {
```

Comment:
I'm sure it's safest to static_cast this as well to make sure the FPU output ends up what you want.